sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusClient.cs

```diff
@@ -326,12 +479,63 @@ public ServiceBusClient(string fullyQualifiedNamespace, TokenCredential credenti
             string subscriptionName,
             ServiceBusProcessorOptions options = default,
             string sessionId = default,
-            CancellationToken cancellationToken = default) =>
-            new ServiceBusProcessor(
+            CancellationToken cancellationToken = default)
+        {
+            if (isConnectionPropertiesValidationNeeded)
```

Comment:
Do we need this bool? I think we should be able to detect if we need to the validation based on whether Connection.ConnectionStringProperties is null.

Reply:
ConnectionStringProperties is a struct. It can never be null.

Reply:
Well we can check the EntityName property on it. If that is null, then no validation is needed.

Reply:
I think we don't even need the ConnectionStringProperties property in ServiceBusConnection class. We can directly check if Connection.EntityPath is not null, then do the validation.